### PR TITLE
Add option to control if the questions is required or not

### LINKED ISSUE #165: Backend: Add a field to indicate if the question is required to be answered.

### PR DESCRIPTION
**Pull request by amrhossamdev**
Fixes #165.

Added a checkbox to control if a question requires a response or not

### CODE REVIEW

**Comment by amrhossamdev** [comment]
My first backend PR
 
image

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
Code changes look great! I'd love to see two tests exercising these new changes: one in the relevant model spec, and one in the relevant request spec to ensure that we don't regress the behavior you've added in the future. LMK if you need clarity on what these tests should look like

**Comment by amrhossamdev** [comment]
> Code changes look great! I'd love to see two tests exercising these new changes: one in the relevant model spec, and one in the relevant request spec to ensure that we don't regress the behavior you've added in the future. LMK if you need clarity on what these tests should look like

Done

File: spec/requests/admins/questions_spec.rb [hunk 1]
```diff
@@ -12,12 +12,13 @@
         sign_in admin
         post admin_study_survey_questions_path(study, survey),
              params: {
-               question: { content_attributes: { type: 'label', label_text: 'Question' } }
+               question: { content_attributes: { type: 'label', required: true, label_text: 'Question' } }
```
Comment (aaronbrethorst) [inline spec/requests/admins/questions_spec.rb:15]: Can you change the value of required to false so we can verify that the value changes through the controller work as expected?

**Review by aaronbrethorst — CHANGES_REQUESTED** [review]
Nice work! One small change remaining and then it’ll be ready to merge.

File: spec/requests/admins/questions_spec.rb (hunk 1 above)
Comment (amrhossamdev) [reply spec/requests/admins/questions_spec.rb:15]: of course

Comment (amrhossamdev) [reply spec/requests/admins/questions_spec.rb]: Done

**Review by aaronbrethorst — APPROVED** [review]
great work 🚀

**Pull request merged**